### PR TITLE
Skip vendor directory when removing generated files

### PR DESCRIPTION
**Pull request by soltysh**
#### What type of PR is this?
/kind bug
/kind cleanup

#### What this PR does / why we need it:
In https://github.com/kubernetes/kubernetes/pull/115198 Tim simplified scripts responsible for removing generated files. All `git_find` implementation spread across 4 files omit vendor and its subdirectories, except the one in `hack/update-codegen.sh` which is causing problems for projects like OpenShift which has its own generated APIs similar to k8s. 

This PR excludes vendor also in that script. 

#### Special notes for your reviewer:
/assign @thockin @liggitt 

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -52,7 +52,7 @@ function git_find() {
     # Similar to find but faster and easier to understand.  We want to include
     # modified and untracked files because this might be running against code
     # which is not tracked by git yet.
-    git ls-files -cmo --exclude-standard "$@"
+    git ls-files -cmo --exclude-standard ':!:vendor/*' "$@"
 }
 
 function git_grep() {
```
Comment (liggitt) [inline hack/update-codegen.sh:58]: same issue in git_grep?

**Comment by liggitt** [comment]
(fix for dependency failure in https://github.com/kubernetes/kubernetes/pull/116948)

File: hack/update-codegen.sh (hunk 1 above)
Comment (soltysh) [reply hack/update-codegen.sh:58]: I think it's less of a problem since we only grep through `--untracked` files.

Comment (thockin) [reply hack/update-codegen.sh:58]: No, `--untracked` means ALSO grep untracked files.

That said, the one place that calls `git_grep` without an explict root path already filters out vendor, so it's not *as* broken.  It seems reasonable to me to move that into `git_grep` and `git_find` - I can't think why this script ever wants to touch vendor/

Comment (soltysh) [reply hack/update-codegen.sh:58]: I've also looked into how the invocation for `git grep`, and pathspecs are only after `--` so we'd need to have the ability to pass arguments before and after the double dash. And just like you're saying that single invocation already handles it as necessary.

Comment (soltysh) [reply hack/update-codegen.sh:58]: Updated.

**Comment by liggitt** [comment]
eyeballing, lgtm, but will let @thockin review and tag

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**